packages/@azure/servicebus/data-plane/lib/session/messageSession.ts

```diff
@@ -587,6 +587,10 @@ export class MessageSession extends LinkEntity {
                 translatedError
               );
               this._notifyError(translatedError);
+            } finally {
+              if (this._receiver) {
+                this._receiver!.addCredit(1);
```

Comment:
We had disabled the  customization of `maxConcurrentCalls` and `maxConcurrentCallsPerSession` options by the user as we didn't understand them then. As part of your investigations, we do understand them now and should bring them back as part of https://github.com/Azure/azure-service-bus-node/issues/179.

Now, with this change in this PR, we are limiting ourselves to only increase the credit by 1. i.e get 1 message, increase credit by 1. This also makes us lose the feature of setting `maxConcurrentCallsPerSession` to anything other than 1.

Reply:
Number of messages the link can hold would still be value set using maxConcurrentCalls.
Each message processing reduces credit by 1, hence restoring the value by adding 1.
Setting it to any other value, say k, would result in increase of link credit by (k * num of messages)

Reply:
> Setting it to any other value, say k, would result in increase of link credit by (k * num of messages)

Yes, thats exactly what I was thinking.

> Number of messages the link can hold would still be value set using maxConcurrentCalls.

Ah! good point. This is because we initially use the value from the `maxConcurrentCallsPerSession` when we `addCredit` for the first time, correct?